# MGST
    Spatial transcriptomics technology enables the systematic acquisition of high-throughput gene expression data on a genome-wide scale while preserving the spatial 
information of tissues in situ, offering significant opportunities to uncover the structural and functional heterogeneity of tissue microenvironments. Existing methods for 
spatial domain identification face limitations in multimodal collaboration and dynamic graph modeling, making it difficult to fully incorporate spatial, expression, and 
morphological information, and often suffer from interference due to high-dimensional noise and data sparsity. In this study, we propose the MGST framework, which leverages a 
multimodal graph fusion strategy to effectively integrate spatial location, gene expression, and tissue morphology information. The framework employs a dynamic adaptive graph 
structure optimization mechanism, overcoming the limitations of traditional graph neural networks that rely on static adjacency relationships, thus enhancing feature expression 
discrimination and generalization performance. Evaluations on multiple datasets (including human DLPFC and breast cancer) demonstrate the superior performance of MGST in 
spatial domain identification. Moreover, MGST not only accurately identifies tumor subtypes associated with immune suppression and metabolic adaptation in breast cancer but 
also reveals functional units regulated by calcium signaling in the mouse forebrain, providing a reliable analytical framework for systematically unveiling the complex 
heterogeneity of tissue microenvironments.
